### ISSUE TITLE
TypeError: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by joelgrodriguez** [colbyfayock/next-wordpress-starter#155]
I just installed a fresh instance of WP and the starter. 

The home page pulls up no problem but...

For some reason when I click on a post I am getting this error. 

Server Error
TypeError: Cannot read property 'length' of undefined

` 
  147 | 
  148 |   const { categories, postId } = post;
> 149 |   const category = categories.length && categories[0];
      |                              ^
  150 |   let { name, slug } = category;
  151 | 
  152 |   return {`

I have not added anything else to the starter (other than the graphql endpoint).
I have not added any posts new posts. 
I added only one additional category for testing. 

Is there a step I am missing?

Here is the WP install: http://newtechmachinery.kinsta.cloud/

thanks

**Comment by colbyfayock** [comment]
hey @joelgrodriguez i was able to reproduce, thats relaly strange

What results do you get when you run this query inside of the GraphiQL IDE?

```
{
  postBy(slug: "hello-world") {
    id
    title
    slug
  }
}
```

the issue is for some reason the query isn't finding the post by it's slug

**Comment by joelgrodriguez** [comment]
This was returned:

`
{
  "data": {
    "postBy": null
  },
  "extensions": {
    "debug": [
      {
        "type": "DEBUG_LOGS_INACTIVE",
        "message": "GraphQL Debug logging is not active. To see debug logs, GRAPHQL_DEBUG must be enabled."
      }
    ]
  }
}
`

**Comment by colbyfayock** [comment]
@doingandlearning have you seen something like this before? any ideas?

the post metadata coming from the homepage has the `hello-world` slug

```
[
  {
    __typename: 'Post',
    author: {
      __typename: 'User',
      avatar: [Object],
      id: 'dXNlcjox',
      name: 'jrodriguez',
      slug: 'jrodriguez'
    },
    id: 'cG9zdDox',
    categories: [ [Object], [Object] ],
    content: '\n' +
      '<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n',
    date: '2021-04-30T02:13:28',
    excerpt: '<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n',
    featuredImage: null,
    modified: '2021-04-30 12:58:57',
    modifiedGmt: '2021-04-30T12:58:57',
    postId: 1,
    title: 'Hello world!',
    slug: 'hello-world',
    isSticky: false
  }
]
```

**Comment by joelgrodriguez** [comment]
I'm thinking it may have something to do with Kinsta?

I just tested this on local install using Local by Flywheel and I am able to load the post with no errors.

**Comment by doingandlearning** [comment]
This is interesting. I think there was some talk about deprecating the `by` functions like `postBy`. It does indeed look like that's happened on your instance. This could be a WP version or WPGraphQL version.

To fix this in our code, we'd move from using `postBy` to using `post`. The new query would look like this:

```
{
  post(id: "hello-world", type: SLUG) {
    id
    title
    slug
  }
}
```

I'll send a message on the WPGraphQL Slack group about it to confirm if a new change has been made.

**Comment by joelgrodriguez** [comment]
I was able to get the expected data back but with a modified query from yours @doingandlearning :

`query MyQuery {
  post(id: "hello-world", idType: SLUG) {
    id
    title
    slug
  }
}`

The query provided above returned an error for me.

Thanks

**Comment by colbyfayock** [comment]
thanks for the heads up both of you, will get that updated to the new query format unless one of you wanna put in a PR :)

**Comment by doingandlearning** [comment]
Very happy to do this. 

Confirmed with the WPGraphQL team that this was an issue. 

![Screenshot_20210430_171508_com Slack](https://user-images.githubusercontent.com/8320213/116723833-cc903e00-a9d7-11eb-99ec-1144fb29e3de.jpg)

**Comment by colbyfayock** [comment]
Fixed via #156
